content/blog/copilot-lessons/index.md

```diff
@@ -0,0 +1,129 @@
+---
+title: "Engineering Lessons from Building Pulumi Copilot"
+date: 2024-12-12T13:56:13-05:00
+draft: false
+meta_desc: Learn key engineering lessons from building Pulumi Copilot, including how to minimize LLM workload, validate outputs, and deal with hallucination.
+meta_image: meta.png
+authors:
+    - artur-laksberg
+    - adam-gordon-bell
+tags:
+    - Copilot
+    - ai
+    - iac
+social:
+    twitter: 🚀 Building Pulumi Copilot taught us key AI engineering insights – Minimize LLM workload, validate outputs rigorously & use "skills" for modular tasks. Plus, sometimes when AI hallucinates features, it reveals exactly what users want! 
+    linkedin: 🚀 Building Pulumi Copilot taught us key AI engineering insights – Minimize LLM workload, validate outputs rigorously & use "skills" for modular tasks. Plus, sometimes when AI hallucinates features (like a "--force" flag), it reveals exactly what users want! 
+---
+**Artur Laksberg is on the Copilot team here at Pulumi. Copilot is an AI assistant for cloud infrastructure. Today he shares key lessons from bringing Copilot to production—including an unexpected discovery about AI hallucinations.**
+
+While reviewing user feedback for Pulumi Copilot, our AI assistant for cloud infrastructure, one message caught my eye: "Your tool doesn't know anything!". Having just made some changes, I braced for the worst. But the evals were still looking strong, so what was going on?
```

Comment:
I do appreciate the "loaded gun" narrative device, but I think a more linear plot would work just fine here.